apps/common/src/main/scala/org/lfdecentralizedtrust/splice/store/UpdateHistory.scala

```diff
@@ -740,15 +740,71 @@ class UpdateHistory(
     domainMigrationInfo.acsRecordTime match {
       case Some(acsRecordTime) =>
         for {
-          _ <- deleteAcsSnapshotsAfter(historyId, previousMigrationId, acsRecordTime)
-          _ <- deleteRolledBackUpdateHistory(historyId, previousMigrationId, acsRecordTime)
+          _ <-
+            if (domainMigrationInfo.synchronizerWasPaused) {
+              for {
+                _ <- verifyNoRolledBackAcsSnapshots(historyId, previousMigrationId, acsRecordTime)
+                _ <- verifyNoRolledBackData(historyId, previousMigrationId, acsRecordTime)
+              } yield ()
+            } else {
+              for {
+                _ <- deleteAcsSnapshotsAfter(historyId, previousMigrationId, acsRecordTime)
+                _ <- deleteRolledBackUpdateHistory(historyId, previousMigrationId, acsRecordTime)
+              } yield ()
+            }
         } yield ()
       case _ =>
         logger.debug("No previous domain migration, not checking or deleting updates")
         Future.unit
     }
   }
 
+  private[this] def verifyNoRolledBackData(
+      historyId: Long, // Not using the storeId from the state, as the state might not be updated yet
+      migrationId: Long,
+      recordTime: CantonTimestamp,
+  )(implicit tc: TraceContext): Future[Unit] = {
+    val action = DBIO
+      .sequence(
+        Seq(
+          sql"""
+            select count(*) from update_history_creates
```

Comment:
the queries here are identical to the `delete` variant so essentially the same query plan as well

```
 Finalize Aggregate  (cost=959598.74..959598.75 rows=1 width=8)
   ->  Gather  (cost=959598.52..959598.73 rows=2 width=8)
         Workers Planned: 2
         ->  Partial Aggregate  (cost=958598.52..958598.53 rows=1 width=8)
               ->  Parallel Index Only Scan using updt_hist_crea_hi_mi_rt on update_history_creates  (cost=0.57..953136.30 rows=2184889 width=0)
                     Index Cond: ((history_id = 2) AND (migration_id = 6) AND (record_time > '1760054400000000'::bigint))
```

Reply:
yeah also the counts are useful in logs imho. So I'd lean towards keeping it as it is.